components/transaction-chart/index.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { Line } from 'react-chartjs-2';
+import { useState, useEffect } from 'react/cjs/react.development';
+import styles from './transaction-chart.module.css';
+const TransactionChart = (props) => {
+  const {
+    transactionDates,
+    creditData,
+    debitData
+  } = props.transactionChartData;
+  const [chartData, setChartData] = useState({});
+  const chart = () => {
+    setChartData({
+      labels: transactionDates,
+      datasets: [
+        {

```

Comment:
Here you can add a improvement, datasets require a array so I can write  a function which generate these objects and inside setChartData you can just call that function. This helps you in two ways : first my function is less cluttered and also add reusability of code.